### PR TITLE
Update compliance matrix for C++

### PR DESCRIPTION
**Pull request by reyang**
Updated the spec compliance matrix to reflect the status of OpenTelemetry C++.

## Changes

Please provide a brief description of the changes here. Update the
`CHANGELOG.md` for non-trivial changes. If `CHANGELOG.md` is updated,
also be sure to update `spec-compliance-matrix.md` if necessary.

Related issues #

Related [oteps](https://github.com/open-telemetry/oteps) #

### CODE REVIEW

**Comment by reyang** [comment]
@jodeev FYI - I cannot add you as a reviewer since you haven't yet joined the OpenTelemetry community yet.
Please go through this https://github.com/open-telemetry/community/blob/master/community-membership.md#member.

File: spec-compliance-matrix.md [hunk 1]
```diff
@@ -12,50 +12,50 @@ status of the feature is not known.
...
-|End                                           |  | +  | +     | +    | +  | +    | + | +  |   | +  |
-|End with timestamp                            |  | +  | +     | +    | +  | +    | + | -  |   | +  |
-|IsRecording                                   |  | +  | +     | +    | +  | +    | + |    |   | +  |
-|Set status                                    |  | +  | +     | +    | +  | +    | + | +  |   | +  |
-|Safe for concurrent calls                     |  | +  | +     |      | +  | +    | + | +  |   | +  |
+|Create root span                              |  | +  | +     | +    | +  | +    | + | +  | + | +  |
+|Create with default parent (active span)      |  | +  | +     | +    | +  | +    | + | +  | + | +  |
+|Create with parent from Context               |  | +  | +     | +    | +  | +    | + | +  | + | +  |
+|Create with explicit parent Span              |  | +  | +     | +    | +  | +    | + | -  | + | +  |
+|Create with explicit parent SpanContext       |  | +  | +     | +    | +  | +    |   | -  | + | +  |
```
Comment (pyohannes) [inline spec-compliance-matrix.md:32]:
```suggestion
|Create with parent from Context               |  | +  | +     | +    | +  | +    | + | +  | - | +  |
|Create with explicit parent Span              |  | +  | +     | +    | +  | +    | + | -  | - | +  |
|Create with explicit parent SpanContext       |  | +  | +     | +    | +  | +    |   | -  | - | +  |
```
As of now, we cannot explicitly specify a span parent, either via Context, Span, or SpanContext.

File: spec-compliance-matrix.md [hunk 2]
```diff
@@ -88,23 +88,23 @@ status of the feature is not known.
...
 |----------------------------------------------|--|----|-------|------|----|------|---|----|---|----|
-|Standard output (logging)                     |  | +  | +     | +    | +  | +    | - | +  |   | +  |
-|In-memory (mock exporter)                     |  | +  | +     | +    | +  | +    | - | -  |   | -  |
+|Standard output (logging)                     |  | +  | +     | +    | +  | +    | - | +  | + | +  |
+|In-memory (mock exporter)                     |  | +  | +     | +    | +  | +    | - | -  | - | -  |
 |[OTLP](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/protocol/otlp.md)|
-|OTLP/gRPC Exporter                            |  | +  | +     | +    |    | +    |   | +  |   | +  |
-|OTLP/HTTP binary Protobuf Exporter            |  | -  | +     | -    |    | +    |   |    |   | +  |
+|OTLP/gRPC Exporter                            |  | +  | +     | +    |    | +    |   | +  | + | +  |
+|OTLP/HTTP binary Protobuf Exporter            |  | -  | +     | -    |    | +    |   |    | + | +  |
```
Comment (pyohannes) [inline spec-compliance-matrix.md:107]:
```suggestion
|OTLP/HTTP binary Protobuf Exporter            |  | -  | +     | -    |    | +    |   |    | - | +  |
```
We only support gRPC for OTLP currently.

File: spec-compliance-matrix.md [hunk 3]
```diff
@@ -12,50 +12,50 @@ status of the feature is not known.
...
-|End with timestamp                            |  | +  | +     | +    | +  | +    | + | -  |   | +  |
-|IsRecording                                   |  | +  | +     | +    | +  | +    | + |    |   | +  |
-|Set status                                    |  | +  | +     | +    | +  | +    | + | +  |   | +  |
-|Safe for concurrent calls                     |  | +  | +     |      | +  | +    | + | +  |   | +  |
+|Create root span                              |  | +  | +     | +    | +  | +    | + | +  | + | +  |
+|Create with default parent (active span)      |  | +  | +     | +    | +  | +    | + | +  | + | +  |
+|Create with parent from Context               |  | +  | +     | +    | +  | +    | + | +  | + | +  |
+|Create with explicit parent Span              |  | +  | +     | +    | +  | +    | + | -  | + | +  |
+|Create with explicit parent SpanContext       |  | +  | +     | +    | +  | +    |   | -  | + | +  |
+|UpdateName                                    |  | +  | +     | +    | +  | +    | + | +  | - | +  |
```
Comment (pyohannes) [inline spec-compliance-matrix.md:33]:
```suggestion
|UpdateName                                    |  | +  | +     | +    | +  | +    | + | +  | + | +  |
```
This is supported, see [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/src/trace/span.cc#L152).

File: spec-compliance-matrix.md [hunk 4]
```diff
@@ -12,50 +12,50 @@ status of the feature is not known.
...
+|Create TracerProvider                         |  | +  | +     | +    | +  | +    | + | +  | + | +  |
+|Get a Tracer                                  |  | +  | +     | +    | +  | +    | + | +  | + | +  |
+|Safe for concurrent calls                     |  | +  | +     |      | +  | +    | + | +  | + | +  |
 |[Tracer](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#tracer-operations)|
-|Create a new Span                             |  | +  | +     | +    | +  | +    | + | +  |   | +  |
-|Get active Span                               |  | +  | +     | +    | +  | +    | + | +  |   | +  |
-|Mark Span active                              |  | +  | +     | +    | +  | +    | + | +  |   | -  |
-|Safe for concurrent calls                     |  | +  | +     |      | +  | +    | + | +  |   | +  |
+|Create a new Span                             |  | +  | +     | +    | +  | +    | + | +  | + | +  |
+|Get active Span                               |  | +  | +     | +    | +  | +    | + | +  | + | +  |
```
Comment (pyohannes) [inline spec-compliance-matrix.md:20]:
```suggestion
|Get active Span                               |  | +  | +     | +    | +  | +    | + | +  | - | +  |
```
I think this refers to the operation on the `Tracer`. We don't currently have this, but we only have the ability to obtain the active span directly from the context (this is cumbersome and will not be the recommended method to obtain the currently active span).

I'd opt to leave this as `-` until we have a `GetActiveSpan` method on the `Tracer`.

**Review by arminru — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@reyang please resolve the conflicts.

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
@reyang Ping ;)

**Comment by reyang** [comment]
@carlosalberto I've resolved the conflicts, sorry for the delay.

**Pull request merged**